@JBludau in the Sycl Trliinos build with [91eee61](https://github.com/kokkos/kokkos/pull/9315/commits/91eee61f91be8cc923c2e34c42e7c9525f04449e) I'm seeing this:

```
/home/ndellin/trilinos/Trilinos-1/kokkos/core/src/impl/Kokkos_Volatile_Load.hpp:41:61: error: use of undeclared identifier 'value'
   41 |            const_cast<std::remove_volatile_t<T>*>(src_ptr), value,
```

--- a/containers/unit_tests/TestUnorderedMap.hpp
+++ b/containers/unit_tests/TestUnorderedMap.hpp
@@ -599,6 +599,52 @@ TEST(TEST_CATEGORY, UnorderedMap_View_as_value) {
   ASSERT_GE(map_copy.capacity(), 200u);
   ASSERT_TRUE(map_copy.is_allocated());
 }
+
+// FIXME_SYCL
+//////////////////////////Tests for UnorderedMap with at type as value type that
+/// does not provide comparison operators
+struct NonComparableType {
+  bool operator==(NonComparableType const &) = delete;
+  bool operator!=(NonComparableType const &) = delete;
+  double dummy[3];  //>8bytes to force triggering the lock-table based atomics
+};
+
+/**
+ * @test This test ensures that an @ref UnorderedMap can be used with a
+ * value_type that does not provide comparison. This is relevant since we need a
+ * SYCL workaround for volatile_load and store to make the map work.
+ */
+TEST(TEST_CATEGORY, UnorderedMap_non_comparable_type_as_value) {
+  using value_type = NonComparableType;
+  using map_type   = Kokkos::UnorderedMap<int, value_type, Kokkos::HostSpace>;
+  map_type map(Kokkos::view_alloc("test NonComparableType umap"), 150);
+  // creation
+  ASSERT_EQ(map.size(), 0u);
+  ASSERT_GE(map.capacity(), 150u);
+  ASSERT_TRUE(map.is_allocated());
+
+  // insert
+  ASSERT_TRUE(map.insert(1, NonComparableType{}).success());
+  ASSERT_TRUE(map.insert(2, NonComparableType{}).success());
+  ASSERT_EQ(map.size(), 2u);
+
+  // copy
+  map_type map_copy(map);
+  ASSERT_EQ(map_copy.size(), 2u);
+  ASSERT_GE(map_copy.capacity(), 150u);
+  ASSERT_TRUE(map_copy.is_allocated());
+
+  // rehash
+  ASSERT_TRUE(map.rehash(200u));
+  ASSERT_GE(map.capacity(), 200u);
+  ASSERT_TRUE(map.is_allocated());
+
+  // assign
+  map_copy = map;
+  ASSERT_EQ(map_copy.size(), 2u);
+  ASSERT_GE(map_copy.capacity(), 200u);
+  ASSERT_TRUE(map_copy.is_allocated());
+}
 }  // namespace Test
 
 #endif  // KOKKOS_TEST_UNORDERED_MAP_HPP

--- a/core/src/impl/Kokkos_Volatile_Load.hpp
+++ b/core/src/impl/Kokkos_Volatile_Load.hpp
@@ -31,15 +31,15 @@ namespace Kokkos {
 template <typename T>
 KOKKOS_FORCEINLINE_FUNCTION T volatile_load(T const volatile* const src_ptr) {
   KOKKOS_IF_ON_HOST(
-      (desul::Impl::device_atomic_fetch_oper(
-           desul::Impl::_load_fetch_operator<T, const T>(),
-           const_cast<std::remove_volatile_t<T>*>(src_ptr), value,
-           desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
+      (return desul::Impl::device_atomic_fetch_oper(
+                  desul::Impl::_load_fetch_operator<T, const T>(),
+                  const_cast<std::remove_volatile_t<T>*>(src_ptr), *src_ptr,
+                  desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
   KOKKOS_IF_ON_DEVICE(
-      (desul::Impl::device_atomic_fetch_oper(
-           desul::Impl::_load_fetch_operator<T, const T>(),
-           const_cast<std::remove_volatile_t<T>*>(src_ptr), value,
-           desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
+      (return desul::Impl::device_atomic_fetch_oper(
+                  desul::Impl::_load_fetch_operator<T, const T>(),
+                  const_cast<std::remove_volatile_t<T>*>(src_ptr), *src_ptr,
+                  desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
 }
 #else
 template <typename T>

--- a/core/src/impl/Kokkos_Volatile_Load.hpp
+++ b/core/src/impl/Kokkos_Volatile_Load.hpp
@@ -33,13 +33,13 @@ KOKKOS_FORCEINLINE_FUNCTION T volatile_load(T const volatile* const src_ptr) {
   KOKKOS_IF_ON_HOST(
       (return desul::Impl::device_atomic_fetch_oper(
                   desul::Impl::_load_fetch_operator<T, const T>(),
-                  const_cast<std::remove_volatile_t<T>*>(src_ptr), *src_ptr,
-                  desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
+                  const_cast<T*>(src_ptr), T(), desul::MemoryOrderRelaxed(),
+                  desul::MemoryScopeDevice());))
   KOKKOS_IF_ON_DEVICE(
       (return desul::Impl::device_atomic_fetch_oper(
                   desul::Impl::_load_fetch_operator<T, const T>(),
-                  const_cast<std::remove_volatile_t<T>*>(src_ptr), *src_ptr,
-                  desul::MemoryOrderRelaxed(), desul::MemoryScopeDevice());))
+                  const_cast<T*>(src_ptr), T(), desul::MemoryOrderRelaxed(),
+                  desul::MemoryScopeDevice());))
 }
 #else
 template <typename T>

